source/gfoidl.Base64/Base64.cs

```diff
@@ -31,6 +31,13 @@ public abstract class Base64 : IBase64
         //---------------------------------------------------------------------
         /// <summary>
         /// Gets the length of the decoded data.
+        /// </summary>     
+        /// <param name="sourceLength">The length of the encoded data.</param>
+        /// <returns>The base64 decoded length of <paramref name="sourceLength" />. Any padding is handled.</returns>
```

Comment:
"Any padding is handled" is only correct for the base64Url encoder, as there is no padding :wink:
For the base64 case, the padding is not considered, hence the result may be off by 2, see https://github.com/gfoidl/Base64/blob/134435f62298df749049dc97ed2465ff76f6ec37/source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Decode.cs#L55

That is the reason why this method is internal, and only the ones that take the base64-Span as input are public, as these handle the padding, see https://github.com/gfoidl/Base64/blob/134435f62298df749049dc97ed2465ff76f6ec37/source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Decode.cs#L32

So what to do?
Following list as it came in my mind:

* rename the method to `GetMaxDecodedLength`
I don't like this options as I wan't to return precise information, and for base64Url it could be precise
* throw a `NotSupportedException` in the base64 case, base64Url is fine
Would need some internal modification, so that base64 case can use the internal method
* Document that in the base64 case the decoded length result may be off by 2
How reads documentation? So this may be somewhat against "least surprise" when the results are off
* rename the method to `GetMaxDecodedLenght` (again)
For base64 document that the result may be off by 2, and it would be better to use the Span-methods.
For base64Url document that the result is exact, as there is no padding.

Do you have any other option?
I have to think about which one I really like most. Until now I went the easy way, and made it `internal`, but I understand that it can be nice to have this `public`, so we'll change this. But how?